telemetry/labels.go

```diff
@@ -395,75 +388,31 @@ func (labels Labels) WithAnonymousGitLabels(workdir string) Labels {
 	return labels
 }
 
-// Define a type for functions that fetch a branch commit
-type fetchFunc func(repo *git.Repository, branch string) (*object.Commit, error)
-
-// Function to fetch from the origin remote
-func fetchFromOrigin(repo *git.Repository, branch string) (*object.Commit, error) {
+func fetchRef(repo *git.Repository, remote string, target string) (*object.Commit, error) {
 	// Fetch from the origin remote
-	cmd := exec.Command("git", "fetch", "--depth", "1", "origin", branch)
-	err := cmd.Run()
-	if err != nil {
-		return nil, fmt.Errorf("error fetching branch from origin: %w", err)
-	}
-
-	// Get the reference of the fetched branch
-	refName := plumbing.ReferenceName(fmt.Sprintf("refs/remotes/origin/%s", branch))
-	ref, err := repo.Reference(refName, true)
-	if err != nil {
-		return nil, fmt.Errorf("error getting reference: %w", err)
-	}
-
-	// Get the commit object of the fetched branch
-	branchCommit, err := repo.CommitObject(ref.Hash())
-	if err != nil {
-		return nil, fmt.Errorf("error getting commit: %w", err)
-	}
-
-	return branchCommit, nil
-}
-
-// Function to fetch from the fork remote
-// GitHub forks are not added as remotes by default, so we need to guess the fork URL
-// This is a heuristic approach, as the fork might not exist from the information we have
-func fetchFromFork(repo *git.Repository, branch string) (*object.Commit, error) {
-	// Get the username of the person who initiated the workflow run
-	username := os.Getenv("GITHUB_ACTOR")
-
-	// Get the repository name (owner/repo)
-	repository := os.Getenv("GITHUB_REPOSITORY")
-	parts := strings.Split(repository, "/")
-	if len(parts) < 2 {
-		return nil, fmt.Errorf("invalid repository format: %s", repository)
-	}
-
-	// Get the server URL: "https://github.com/" in general,
-	// but can be different for GitHub Enterprise
-	serverURL := os.Getenv("GITHUB_SERVER_URL")
-
-	forkURL := fmt.Sprintf("%s/%s/%s", serverURL, username, parts[1])
-
-	cmd := exec.Command("git", "remote", "add", "fork", forkURL)
+	output := bytes.Buffer{}
+	cmd := exec.Command("git", "fetch", "--depth", "1", remote, target)
+	cmd.Stdout = &output
+	cmd.Stderr = &output
 	err := cmd.Run()
 	if err != nil {
-		return nil, fmt.Errorf("error adding fork as remote: %w", err)
+		return nil, fmt.Errorf("error fetching branch from origin: %w\n%s", err, output.String())
 	}
+	output.Reset()
 
-	cmd = exec.Command("git", "fetch", "--depth", "1", "fork", branch)
+	// Get the commit hash of the fetched remote
+	cmd = exec.Command("git", "rev-parse", "FETCH_HEAD")
```

Comment:
Would this work?:

```suggestion
	cmd = exec.Command("git", "rev-parse", target)
```

Since this runs on the host, just trying to avoid any funny business with parallel fetches.